mindsdb/integrations/handlers/intercom_handler/intercom_handler.py

```diff
@@ -0,0 +1,130 @@
+            access_token = args['access_token']
+        self._headers = {
+            "Accept": "application/json",
+            "Authorization": f"Bearer {access_token}"
+        }
+
+        _tables = [
+            Articles,
+        ]
+
+        for Table in _tables:
+            self._register_table(Table.name, Table(self))
+
+    def check_connection(self) -> StatusResponse:
+        """checking the connection
+
+        Returns:
+            StatusResponse: whether the connection is still up
+        """
+        response = StatusResponse(False)
+
+        try:
+            self.connect()
+            response.success = True
+        except Exception as e:
+            log.logger.error(f'Error connecting to Intercom API: {e}!')
+            response.error_message = e
+
+        self.is_connected = response.success
+        return response
+
+    def connect(self) -> StatusResponse:
+        """making the connectino object
+        """
+        if self.is_connected and self.connection:
+            return self.connection
+
+        if self._headers:
+            try:
+                response = requests.get(
+                    url=self._baseUrl,
+                    headers=self._headers
+                )
+                if response.status_code == 200:
+                    self.connection = response
+                    self.is_connected = True
+                    return StatusResponse(True)
+                else:
+                    raise Exception(f"Error connecting to Intercom API: {response.status_code} - {response.text}")
+            except requests.RequestException as e:
+                raise Exception(f"Request to Intercom API failed: {str(e)}")
+
+        raise Exception("Access token is missing")
+
+    def native_query(self, query: str) -> StatusResponse:
+        """Receive and process a raw query.
+
+        Parameters
+        ----------
+        query : str
+            query in a native format
+
+        Returns
+        -------
+        StatusResponse
+            Request status
+        """
+        ast = parse_sql(query, dialect="mindsdb")
+        return self.query(ast)
+
+    def call_intercom_api(self, endpoint: str, method: str = 'GET', params: dict = None, data=None) -> pd.DataFrame:
+        if not params:
+            params = {}
+
+        url = f"{self._baseUrl}{endpoint}"
+
+        if method.upper() in ('GET', 'POST', 'PUT', 'DELETE'):
+
+            if method.upper() in ('POST', 'PUT', 'DELETE'):
+                response = requests.request(method, url, headers=self._headers, params=params, json=json.loads(data))
+            else:
+                response = requests.get(url, headers=self._headers, params=params)
+
+            if response.status_code == 200:
+                data = response.json()
+                return pd.DataFrame([data])
+            else:
+                raise Exception(f"Error connecting to Intercom API: {response.status_code} - {response.text}")
```

Comment:
We can use the requests response.raise_for_status()

Reply:
Done